src/json/def_to_json.cr

```diff
@@ -0,0 +1,177 @@
+module JSON
+  # The `def_to_json` macro generates a `to_json(json : JSON::Builder)` method based on provided *mappings*.
+  #
+  # It is a lightweight alternative to `JSON.mapping` if you don't need to declare instance variables and a parser.
+  #
+  # The generated method invoks `to_json(JSON::Builder)` on each of the values returned
+  # by `self`'s properties from *mappings*, or - if a converter is specified - `to_json(value, JSON::Builder)` on the converter.
+  #
+  # ### Example
+  #
+  # ```
+  # require "json"
+  #
+  # record Location, lat : Float64, long : Float64 do
+  #   JSON.def_to_json([lat, long])
+  # end
+  #
+  # record House, street : String, street_number : Int32, location : Location do
+  #   JSON.def_to_json(
+  #     address: true,
+  #     loc: {property: location},
+  #     empty_field: {emit_null: true},
+  #   )
+  #
+  #   def address
+  #     "#{street} #{street_number}"
+  #   end
+  #
+  #   def empty_field
+  #     nil
+  #   end
+  # end
+  #
+  # house = House.new("Crystal Road", 1234, Location.new(12.3, 34.5))
+  # house.to_json # => %({"address":"Crystal Road 1234","loc":{"lat":12.3,"long":34.5},"empty_field":null})
+  # ```
+  #
+  # ### Usage
+  #
+  # `JSON.def_to_json` must receive a series of named arguments, or a named tuple literal, or a hash literal,
+  # whose keys will define JSON properties.
+  #
+  # The value of each key can either be `true` or a hash or named tuple literal with the following options:
+  # * **property**: the property name on the Crystal object (as opposed to the key in the JSON document)
```

Comment:
Hmm I'm not sure I completely agree, but it makes sense, and given this concept difference I think it's acceptable (and interesting to have too!).

Also I think you should copy/paste this discussion (only the important bits: question & answer?) in the issue's conversation, so your explaination about the differences of concept can't get lost as a line comment when/if you change this particular line of code.